### PR TITLE
DataGrid: Allow to set the '0000' number format in 'onXlsxCellPrepared'

### PR DESCRIPTION
**Pull request by IgnatovDan**

### CODE REVIEW

File: js/client_exporter/xlsx/xlsx_cell_format_helper.js [hunk 1]
```diff
@@ -7,7 +7,7 @@ const xlsxCellFormatHelper = {
         let result = null;
         if(typeUtils.isDefined(sourceObj)) {
             let numberFormatId;
-            if(typeUtils.isDefined(sourceObj.numberFormat) && typeUtils.isNumeric(sourceObj.numberFormat)) {
+            if(typeUtils.isDefined(sourceObj.numberFormat) && typeof sourceObj.numberFormat === 'number') {
```
Comment (jsdmitry) [inline js/client_exporter/xlsx/xlsx_cell_format_helper.js:10]: You can use the `isNumber = typedUtils.isNumeric,` for check a number type.

Comment (IgnatovDan) [reply js/client_exporter/xlsx/xlsx_cell_format_helper.js:10]: The '0000' format string formats Number(42) to '0042' string.
The typedUtils.isNumeric('0000') returns true, '0000' is serialized as numFmtId="0000" so the 'Generic' format (its 'id' is '0') is used and Number(42) is shown as '42' string.

**Review by jsdmitry — APPROVED** [review]

File: js/client_exporter/xlsx/xlsx_cell_format_helper.js (hunk 1 above)
Comment (churkin) [reply js/client_exporter/xlsx/xlsx_cell_format_helper.js:10]: Why we need `typeUtils.isDefined(sourceObj.numberFormat)` check if the `typeof sourceObj.numberFormat === 'number'` returns `false` for `null` and `void 0`?

Comment (IgnatovDan) [reply js/client_exporter/xlsx/xlsx_cell_format_helper.js:10]: IsDefined will be removed in https://github.com/DevExpress/DevExtreme/pull/5724

**Pull request merged**